### PR TITLE
adds signing and verifying messages to the cbridge

### PR DESCRIPTION
**Pull request by QuestofIranon**

### CODE REVIEW

**Comment by QuestofIranon** [comment]
see also hashgraph/hedera-sdk-c#6

File: src/bridge/key.rs [hunk 1]
```diff
@@ -37,6 +38,21 @@ pub unsafe extern "C" fn hedera_secret_key_from_str(s: *const c_char, out: *mut
...
 }
 
+#[doc(hidden)]
+#[no_mangle]
+pub unsafe extern "C" fn hedera_secret_key_sign(p: *mut SecretKey, message: *const c_void, message_len: usize, out: *mut Signature) -> u64 {
+    debug_assert!(!message.is_null());
+    debug_assert!(!p.is_null());
+    debug_assert!(!out.is_null());
+
+    let message = message as *const u8;
```
Comment (mehcode) [inline src/bridge/key.rs]: Just have the param type be `*const u8`

File: src/key.rs [hunk 2]
```diff
@@ -479,6 +479,12 @@ impl Signature {
             bytes.as_ref(),
         )?))
     }
+
+    /// Format a `Signature` as a vec of bytes.
+    pub fn to_bytes(&self) -> Vec<u8> {
```
Comment (mehcode) [inline src/key.rs]: Why do we need this?

**Review by mehcode — CHANGES_REQUESTED** [review]

**Comment by QuestofIranon** [comment]
Should be good now

**Pull request merged**